### ISSUE TITLE
No Approve button for translations

### ISSUE THREAD

**Issue by kdmatrosov** [Codewars/codewars.com#1606]
There is [Kotlin Translation](https://www.codewars.com/kumite/5bc831e7c3ba465d0e000146?sel=5bc831e7c3ba465d0e000146) and there is no **Approve** button. More over there are other 10 translations to Kotlin from me without button

**Comment by error256** [comment]
The author of the kata is active. I think the buttons are now supposed to be shown only to those who can use them, so it's correct?

**Comment by FArekkusu** [comment]
The author is active, so nobody except him can approve it right now.

> Tells @error256 than Unnamed can't see the button.

This is quite funny :)

**Comment by kdmatrosov** [comment]
I supposed that anyone with 4k+ honor can approve due to [this](https://github.com/Codewars/codewars.com/wiki/Privileges)

**Comment by kazk** [comment]
It should be shown if you're the author of the kata; or you have enough honor to coauthor kata and the translation was published more than a week ago or the kata author has been inactive over a month.

There was a bug that made those buttons to always show even if it won't do anything and I fixed it last week. I might revert to always showing them since the cache is still acting weird.

**Comment by kdmatrosov** [comment]
> you have enough honor to coauthor kata and the translation was published more than a week ago or the kata author has been inactive over a month

Thanks. I think there is the problem of not approved translations. For several weeks I did some translations. Regularly I want to start new translation and find out not approved translations which are in review for year and more. They are normal. Work as expected, have random tests. Author of kata can forget about them and notifications are shown only once. So we lose a lot of new translations. Maybe you can make some voting system as on stackoverflow? You have reputation and can vote normal or not post of user

**Comment by kazk** [comment]
> Regularly I want to start new translation and find out not approved translations which are in review for year and more.

Do you remember which kata had this? Admins have a page for list of all pending translations and the oldest one I see there is 3 days old. This might be another issue :man_facepalming:

Maybe we can add something to show the number of pending translations so power users can find them.

**Comment by kdmatrosov** [comment]
> Do you remember which kata had this? 

https://www.codewars.com/kata/convert-a-number-to-a-string/translations C
https://www.codewars.com/kata/vowel-count/translations F# Swift Rust...
https://www.codewars.com/kata/basic-mathematical-operations/translations Go, Elixir

And there are others, I need time to find them all. One more problem with old katas/translation is no clear action to "UP" your theme. For example I used to spam a little to attract attention

**Comment by kazk** [comment]
@kdmatrosov thanks! I'll look into what's going on with them.

![image](https://user-images.githubusercontent.com/639336/47387250-9ca7a780-d6c3-11e8-855c-7970d95fae41.png)

4 years :man_facepalming:

**Comment by kazk** [comment]
Wait, that kata already has Haskell.

**Comment by kdmatrosov** [comment]
@kazk I pointed languages which wait for a long

**Comment by kazk** [comment]
Yeah, I scrolled down to "closed translations" without noticing. Thanks again.

**Comment by kazk** [comment]
"Convert a Number to a String!" C++ translation and "Vowel Count" F# translation has a comment asking for change.

The test output of "Basic Mathematical Operations" Elixir translation seems incomplete. Go translation doesn't seem completed? At least it's missing random tests.

I'm still not sure why those don't show up on admin's pending translations list, but those needs to be fixed before approval.

**Comment by kazk** [comment]
> I'm still not sure why those don't show up on admin's pending translations list

Found that it's paginated (25 items each), but it's broken.


I'll revert the buttons for now since I have no idea how the caching is getting in the way.

**Comment by kazk** [comment]
Buttons should be shown for published translations.
I had to manually invalidate cache so please let me know if there's one that I missed.

**Comment by kdmatrosov** [comment]
> "Vowel Count" F# translation has a comment asking for change.
But this comment was 3 months ago and translation was published 11 months ago. Translation's author could miss notification
> The test output of "Basic Mathematical Operations" Elixir translation seems incomplete. Go translation doesn't seem completed? At least it's missing random tests.
But there is no feedback in translations. How can author know what to do without it?

**Comment by kazk** [comment]
> But this comment was 3 months ago and translation was published 11 months ago.
> Translation's author could miss notification

Doesn't translator get notification when someone comments?

> There is no feedback in translations. How can author know what to do without it?

Someone needs to review and give feedback. Pending translations are difficult to discover and that's part of the problem.

**Comment by kdmatrosov** [comment]
One more [kata with pending translations](https://www.codewars.com/kata/permutations/translations)

If you create some theme I will post such links there

**Comment by kdmatrosov** [comment]
[Pending for 3 years](https://www.codewars.com/kumite/55c047b2c56a69bd5c000039?sel=55c047b2c56a69bd5c000039)

**Comment by kazk** [comment]
@kdmatrosov please use https://gitter.im/Codewars/codewars.com for that. This issue is about the buttons.
